feat: wit for arbit qubits

vprusso opened the pull request:
Closes: #609 

Additionally, this MR fixes the following warnings:

```
=========================================== warnings summary ===========================================
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
tests/unit/benchmarks/test_mirror_circuits.py::TestMirrorCircuitGeneration::test_generate_mirror_circuit
  /home/vrusso/Projects/metriq-gym/metriq_gym/benchmarks/mirror_circuits.py:335: DeprecationWarning: Treating CircuitInstruction as an iterable is deprecated legacy behavior since Qiskit 1.2, and will be removed in Qiskit 3.0. Instead, use the `operation`, `qubits` and `clbits` named attributes.
    for instr, qargs, _ in pauli_layer.data:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

vprusso committed:
feat: wit for arbit qubits

vprusso committed:
fix: update notebook output

cosenal commented on metriq_gym/schemas/wit.schema.json:
```diff
@@ -5,25 +5,80 @@
+        "interaction_coupling_strength": {
+            "type": "number",
+            "description": "Interaction coupling strength (g) applied during the two-sided coupling step.",
+            "default": 1.5707963267948966,
```
I am guessing those values are fractions of pi? It's odd that one can't use symbols in here. One more reason to get rid of those parameters.

vprusso replied:
Good point. Those have since been removed (see the above comment/response).

cosenal reviewed (commented):
Some preliminary comments

vprusso committed:
chore: merge main

vprusso committed:
chore: merge main

vprusso committed:
chore: simplify wit benchmark

vprusso committed:
chore: simplify wit benchmark

vprusso committed:
Merge remote-tracking branch 'origin/main' into 609-generalize-wit-benchmark

# Conflicts:
#	docs/source/cli_workflows.rst
#	docs/source/end_to_end_tutorial.ipynb
#	docs/source/quickstart.rst
#	metriq_gym/benchmarks/mirror_circuits.py
#	metriq_gym/benchmarks/wit.py
#	uv.lock

vprusso committed:
fix: set WIT n_qubits_per_side minimum to 2

The generalized WIT config requires the message register (message_size >= 1)
to be strictly smaller than n_qubits_per_side, so a single qubit per side is
invalid. The schema previously advertised a minimum of 1, which caused the
e2e benchmark test (which shrinks each config value to its schema minimum) to
build an invalid n_qubits_per_side=1 configuration and raise ValueError during
dispatch.
